torch/_inductor/codegen/wrapper.py

```diff
@@ -334,6 +335,32 @@ def call(args):
                 )
             V.graph.sizevars.codegen(self.wrapper_call, V.graph.graph_inputs)
 
+    def write_onednn_param_gen_utils(self):
+        self.header.splice(
+            """
+            onednn_param_cache = dict()
+            def get_onednn_conv_param(op_id, src, weight, bias, padding, stride, dilation, groups, attr, scalars, algo):
+                if op_id in onednn_param_cache:
+                    conv_param = onednn_param_cache[op_id]
+                else:
+                    conv_param = torch.ops.mkldnn._conv_param_generation(src, weight, bias, padding, stride, dilation,
+                                                                         groups, attr, scalars, algo)
+                    onednn_param_cache[op_id] = conv_param
```

Comment:
How would these handles be deleted, e.g., wrapping it to a Python object and call delete from its `__del__` function?

Reply:
Yes, that is exactly the open I raised in the PR comments that would like to discuss with you. In current TorchDynamo/TorchInductor design, the TorchInductor generated python wrapper will be parsed as a context and keeps alive until the entire process exit. Even we attach an exit mechanism seems it will not be triggered. Do we still need to add a mechanism to do that?

Reply:
I think the feature itself should do its job right, like having a way to get its handles disposed. Then it is wrapper's job to properly manage the lifecycle with this contract. The latter could be another design if it is missing now...

Reply:
I've added mechanism for deleting these handles, and also added a config to control the param caching (with default to be False)